src/utils/daos_metrics/daos_metrics.c

```diff
@@ -199,27 +267,50 @@ main(int argc, char **argv)
 		extra_descriptors |= D_TM_INCLUDE_TYPE;
 
 	if (format == D_TM_CSV)
-		d_tm_print_field_descriptors(extra_descriptors, stdout);
+		filter &= ~D_TM_DIRECTORY;
+	else
+		filter |= D_TM_DIRECTORY;
 
-	while ((num_iter == 0) || (iteration < num_iter)) {
-		d_tm_iterate(ctx, root, 0, filter, NULL, format, extra_descriptors,
-			     ops, stdout);
-		iteration++;
-		sleep(delay);
-		if (format == D_TM_STANDARD)
-			printf("\n\n");
+	if (strlen(jobid) > 0) {
+		struct d_tm_node_t *root;
+		struct d_tm_context *ctx;
+		struct iter_pid_arg iter_arg;
+
+		/* fetch metrics from client side by jobid */
+		ctx = d_tm_open(DC_TM_JOB_ROOT_ID);
+		if (!ctx) {
+			printf("Unable to find job %s\n", jobid);
+			exit(0);
+		}
+
+		root = d_tm_find_metric(ctx, jobid);
+		if (root == NULL) {
+			d_tm_close(&ctx);
+			printf("Unable to find job %s\n", jobid);
+			exit(0);
+		}
+
+		iter_arg.dirname = dirname;
+		iter_arg.filter = filter;
+		iter_arg.iter_cb = iter_cb;
+		iter_arg.delay = delay;
+		iter_arg.num_iter = num_iter;
+		d_tm_iterate(ctx, root, 0, D_TM_DIRECTORY, NULL, format, extra_descriptors,
```

Comment:
I played around with the PR on my development node where I'm running the server and some client processes. When I run daos_metrics -j <jobid>, I get the engine metrics mixed in and the client metrics are all zero. I wonder if it's because the wrong context is being used in the iterator?

Reply:
did you provide correct jobid after -j ?  hmm, the iteration will iterate all pid under root/jobid, then iterate each pid metrics here. see d_tm_iterate->iter_per_pid->process_metrics().  It should only include client object metrics for the moment. 

I tried this on my side with multiple jobid, it works here.